File: backend/routes/productos.routes.js
const express = require("express");
const router = express.Router();

const productosController = require("../controllers/productos.controller");
const { verificarToken, verificarAdmin } = require("../middleware/auth.middleware");

// obtener productos
router.get("/", productosController.obtenerProductos);

// crear producto
router.post("/", verificarToken, verificarAdmin, productosController.crearProducto);

// eliminar producto
router.delete("/:id", verificarToken, verificarAdmin, productosController.eliminarProducto);


// Actualizar producto (Editar)
router.put("/:id", verificarToken, verificarAdmin, productosController.actualizarProducto);


module.exports = router;

File: backend/routes/productos_routes.py
from fastapi import APIRouter, Depends
from controllers import productos_controller
from schemas.productos_schema import Producto
from middleware.auth_middleware import verificarAdmin

router = APIRouter()


@router.get("/")
def get_productos():
    return productos_controller.obtenerProductos()


@router.post("/")
def create_producto(producto: Producto, user=Depends(verificarAdmin)):
    productos_controller.crearProducto(producto)
    return {"mensaje": "Producto creado"}


@router.delete("/{id}")
def delete_producto(id: int, user=Depends(verificarAdmin)):
    productos_controller.eliminarProducto(id)
    return {"mensaje": "Producto eliminado"}


@router.put("/{id}")
def update_producto(id: int, producto: Producto, user=Depends(verificarAdmin)):
    productos_controller.actualizarProducto(id, producto)
    return {"mensaje": "Producto actualizado"}